### ISSUE TITLE
mods in game config settings?

### ISSUE THREAD

**Issue by vincentropy** [arenarium/battleground_core#102]
@phylyc Should we specify mods in the settings dict in the game config?
That way games with mods are treated on an equal footing to games without mods.

**Comment by phylyc** [comment]
yeah, we can do so. Then you would just need to change the whole paragraph in the site_runner.game_engine_factory to check for "mods" in game_config["settings"] and access this dict.

**Comment by phylyc** [comment]
Actually, I was placing the mods dict not within the setting entry to not hand the game engine init method the mod settings on instantiation of the game engine. Doing so would require us to always add **kwargs to the init functions, because it could be that any game has the mods entry in its settings, but this entry is only important pre-instantiation (in the current set-up). It is possible to restructure it, though it opens the possibility for instantiations of game engines that do not handle all the specifications in the settings. Right now it would throw an error. I thought it would be nicer for debugging, but I can change it.

**Comment by phylyc** [comment]
Idea: remove knowledge of mods in site_runner.game_engine_factory and introduce a wrapper class that takes care of the modding.
